### PR TITLE
feat: format token prices by significant digits

### PR DESCRIPTION
**Pull request by 2wheeh**
## Background

https://www.notion.so/delight-labs/3677d06c9eda80dbbbbcf11d39622b0b?source=copy_link

## Description
- Add `formatTokenPrice` util: 2 decimals for ≥$1, 4 significant digits for <$1, `< $0.000001` floor.
- Add `TokenPriceFormatter` component wrapping it with a full-value hover tooltip.
- Apply to token price displays only: Analytics All Tokens, Top Movers, and token detail. Volume/TVL/Fee are unchanged.
- Small prices now show real values (e.g. `$0.001234`) instead of a flat `< $0.01`.
- Add unit tests covering every spec example plus boundary, floor, zero, and compact cases.

## Checklist

- [x] token price displays by the new policy
- [x] the others keep the existing behavior

## Summary by CodeRabbit

* **New Features**
  * Added improved token price formatting across analytics and token detail views for clearer display of very small and large values.
  * Introduced tooltip support for truncated token prices so the full value is easier to inspect.

* **Bug Fixes**
  * Updated token price displays to use consistent formatting in token lists, top movers, mobile token cards, and token summaries.

* **Tests**
  * Added coverage for token price formatting across zero, small, standard, and large value ranges.

### CODE REVIEW

File: src/pages/Analytics/AllTokens.tsx [hunk 1]
```diff
@@ -245,7 +246,7 @@ function AllTokens({ whiteList }: { whiteList?: string[] }) {
                 width: 190,
                 hasSort: true,
                 render(price) {
-                  return price && <CurrencyFormatter value={price} />;
+                  return price && <TokenPriceFormatter value={price} />;
```
[the remark this answers is not in the document]
Comment (2wheeh) [reply src/pages/Analytics/AllTokens.tsx:249]: price is a string, so 0 arrives as `"0"` (truthy) and already renders `$0`. Falsy values stay blank by design (avoids a misleading $0 for missing data). Safe as-is.

**Comment by 2wheeh** [comment]
@coderabbitai review

**Review by winnieworld — APPROVED** [review]
